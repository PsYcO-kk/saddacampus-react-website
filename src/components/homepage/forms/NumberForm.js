import React from "react";

export default class NumberForm extends React.Component {
	constructor(props){
		super(props);
		this.state = {
			number: '',
			error: ''
		};
	}
	validateInput = () => {
		return this.state.number.match(/^\d{10}$/);
	}
	handleNumberInput = (e) => {
		const number = e.target.value;
		if(!number || number.match(/^\d{1,10}$/)){
			this.setState(() => ({ error: '', number: number }));
		}
	}
	changeView = (message , country_code) => {
		const newState = {
			NumberForm: false,
			OtpForm: true,
			ProfileSettings: false,
      number: this.state.number,
      country_code : country_code,
			otp: '',
			message: message
		}
		this.props.changeView(newState);
	}
	handleSubmit = (e) => {
		e.preventDefault();
		const country_code = e.target.country_code.value;
		const number = this.state.number;
		if(this.validateInput()){
			fetch('http://127.0.0.1:3000/v1/membership/otp/'+country_code+'/'+number)
			.then((response) => response.json())
			.then((data) => {
				if(data.success){
					this.setState(() => ({ error: '' }));
					this.changeView(data.message,country_code);
				}
				else{
					this.setState(() => ({ error: data.message }));
				}
			})
			.catch((error) => {
				this.setState(() => ({ error }));
			});
		}
		else{
			document.getElementsByClassName('alert').className += ' alert-danger';
			this.setState(() => ({
				error: 'Please enter a valid mobile number.'
			}));
		}
	}
    render(){
        return(
            <form onSubmit={this.handleSubmit}>
				{this.state.error && <div className="alert alert-danger">{this.state.error}</div>}
				<label htmlFor="numberInput">Enter your mobile number to continue:</label>
				<div className="input-group mb-3">
					<select className="custom-select form-control" defaultValue="+91" name="country_code">
						<option value="+91">+91</option>
					</select>
					<input type="tel" className="form-control" name="numberInput" value={this.state.number} onChange={this.handleNumberInput} placeholder="# 7777-777777" />
					<div className="input-group-append">
						<button className="btn btn-outline-secondary" type="submit">Go</button>
					</div>
				</div>
				<small className="form-text text-muted">Don't worry. We won't stalk you.</small>
			</form>
        );
    }
}