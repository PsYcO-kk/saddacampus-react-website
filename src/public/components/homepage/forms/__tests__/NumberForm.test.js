import React from 'react';
import { shallow } from 'enzyme';
import NumberForm from '../NumberForm';

let wrapper;

beforeEach(() => {
	wrapper = shallow(<NumberForm />);
});

test('should render NumberForm correctly', () => {
	expect(wrapper).toMatchSnapshot();
});

test('should set number on valid input', () => {
	const value = '7465465475';
	wrapper. find('input').simulate('change', {
		target: { value }
	});
	expect(wrapper.state('number')).toBe(value);
});

test('should not set number on invalid input', () => {
	const value = '7465465475bff';
	wrapper. find('input').simulate('change', {
		target: { value }
	});
	expect(wrapper.state('number')).toBe('');
});

test('should set and render error on invalid number submit', () => {
	const country_code = '+91';
	const number = '765765';
	wrapper. find('form').simulate('submit', {
		preventDefault: () => {},
		target: {
			country_code: { value: country_code },
			numberInput: { value: number }
		}
	});
	expect(wrapper.state('error')).toBe('Please enter a valid mobile number.');
	expect(wrapper).toMatchSnapshot();
});
