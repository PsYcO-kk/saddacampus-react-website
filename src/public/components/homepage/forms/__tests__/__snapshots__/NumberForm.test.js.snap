// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NumberForm correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="numberInput"
  >
    Enter your mobile number to continue:
  </label>
  <div
    className="input-group mb-3"
  >
    <select
      className="custom-select form-control"
      defaultValue="+91"
      id="country_code"
      name="country_code"
      required={true}
    >
      <option
        value="+91"
      >
        +91
      </option>
    </select>
    <input
      className="form-control"
      id="numberInput"
      name="numberInput"
      onChange={[Function]}
      placeholder="# 7777-777777"
      required={true}
      type="tel"
      value=""
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-outline-secondary"
        disabled={false}
        type="submit"
      >
        Go
      </button>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Don't worry. We won't stalk you.
  </small>
</form>
`;

exports[`should set and render error on invalid number submit 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="alert alert-danger"
  >
    Please enter a valid mobile number.
  </div>
  <label
    htmlFor="numberInput"
  >
    Enter your mobile number to continue:
  </label>
  <div
    className="input-group mb-3"
  >
    <select
      className="custom-select form-control"
      defaultValue="+91"
      id="country_code"
      name="country_code"
      required={true}
    >
      <option
        value="+91"
      >
        +91
      </option>
    </select>
    <input
      className="form-control"
      id="numberInput"
      name="numberInput"
      onChange={[Function]}
      placeholder="# 7777-777777"
      required={true}
      type="tel"
      value=""
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-outline-secondary"
        disabled={false}
        type="submit"
      >
        Go
      </button>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Don't worry. We won't stalk you.
  </small>
</form>
`;
